server/src/internalClusterTest/java/org/opensearch/action/search/TransportSearchIT.java

```diff
@@ -370,6 +371,51 @@ public void testSearchIdle() throws Exception {
         });
     }
 
+    public void testSearchIdleWithSegmentReplication() throws Exception {
+        int numOfReplicas = 1;
+        internalCluster().ensureAtLeastNumDataNodes(numOfReplicas + 1);
+        final Settings.Builder settings = Settings.builder()
+            .put(IndexMetadata.SETTING_NUMBER_OF_SHARDS, randomIntBetween(1, 5))
+            .put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, numOfReplicas)
+            .put(IndexMetadata.SETTING_REPLICATION_TYPE, ReplicationType.SEGMENT);
+        assertAcked(prepareCreate("test").setSettings(settings).setMapping("created_date", "type=date,format=yyyy-MM-dd"));
+        ensureGreen("test");
+        assertAcked(
+            client().admin()
+                .indices()
+                .prepareUpdateSettings("test")
+                .setSettings(
+                    Settings.builder()
+                        .put(IndexSettings.INDEX_SEARCH_IDLE_AFTER.getKey(), TimeValue.timeValueMillis(randomIntBetween(50, 500)))
+                )
+        );
+
+        assertBusy(() -> {
```

Comment:
Why do you have to poll just to ultimately assert on an index setting?

Reply:
Ah thanks for the callout will remove this.  I was originally validating against the `isSearchIdle` while adding more logic to that function so needed to wait for the idle state.  Now that we split to a separate function the wait is not required.

Reply:
updated.